mtags/src/main/scala-3/scala/meta/internal/pc/completions/CompletionSuffix.scala

```diff
@@ -8,10 +8,15 @@ package scala.meta.internal.pc.completions
  */
 case class CompletionSuffix(
     brace: Boolean,
-    bracket: Boolean,
+    bracket: HasBracket,
     template: Boolean,
     snippet: SuffixKind,
```

Comment:
We already have `SuffixKind.Bracket` do we actually need the other parameter? We could just have `parameterCount`, no? Or the enums could have the count parameter

Reply:
This is are two a bit different things. `snippet` is what we actually add, and all the others are all the existing endings. But as @jkciesluk noticed, we might have a `bracket` suffix but not add anything as a snippet. We could do:
```
case class CompletionSuffix(suffixes: Set[SuffixKind], snippet: SuffixKind)
```
and add param count to enums but I doesn't really make much sense for e.g. `template`.

Reply:
Ach alright, I would change it and maybe uset have parameter count as an Option?

Reply:
But change to option inside of the `HasBracket` class or drop the class? 'cause if the second one I think I agree with @jkciesluk

Reply:
I was thinking of something like:
```
enum SuffixKind(paramsCount: Option[Int]):
  case Brace(count: Int) extends SuffixKind(Some(count))
  case Bracket (count: Int) extends SuffixKind(Some(count))
  case Template extends SuffixKind(None)
  case NoSuffix extends SuffixKind(None)
```

Reply:
I would want to avoid:
```
    brace: Boolean,
    bracket: HasBracket,
    template: Boolean,
```
where it's unclear why some params are a Boolean and some arent. And what do they even mean.

Reply:
It would still have to be smth like:
```
case class CompletionSuffix(suffixes: Set[SuffixKind], snippet: SuffixKind)
```
since we can't just have a single suffix kind

Reply:
> enum SuffixKind(paramsCount: Option[Int]):
  case Brace(count: Int) extends SuffixKind(Some(count))
  case Bracket (count: Int) extends SuffixKind(Some(count))
  case Template extends SuffixKind(None)
  case NoSuffix extends SuffixKind(None)
  
  Why not just 
  ```
  enum SuffixKind:
    case Brace(count: Int) extends SuffixKind
    case Bracket (count: Int) extends SuffixKind
    case Template extends SuffixKind
    case NoSuffix extends SuffixKind
 ```
 
?

Reply:
> It would still have to be smth like:
> 
> ```
> case class CompletionSuffix(suffixes: Set[SuffixKind], snippet: SuffixKind)
> ```
> 
> since we can't just have a single suffix kind

I can imagine doing somethink like 
```scala
  case class CompletionSuffix(suffixes: List[SuffixKind], snippet: SuffixKind)
  
  def toEdit(suffix: CompletionSuffix): String = {
    def loop(suffixes: List[SuffixKind]): String = {
      if (suffixes.isEmpty) ""
      else {
        val snippet = if suffixes.head == suffix.snippet then "$0" else ""
        suffixes.head match {
          case SuffixKind.Brace(count) => s"[$snippet]" + loop(suffixes.tail)
          case SuffixKind.Bracket(count) => s"($snippet)" + loop(suffixes.tail)
          case SuffixKind.Template => s"{$snippet}" + loop(suffixes.tail)
          case SuffixKind.NoSuffix => ""
        }
      }
    }
    loop(suffix.suffixes)
  }
  ```